tests/ui/treat-err-as-bug/eagerly-emit.stderr

```diff
@@ -0,0 +1,26 @@
+error: `report_selection_error` did not emit an error
+  --> $DIR/eagerly-emit.rs:7:11
+   |
+LL | fn f() -> impl Foo {
+   |           ^^^^^^^^
+
+error[E0277]: the trait bound `i32: Foo` is not satisfied
+  --> $DIR/eagerly-emit.rs:7:11
+   |
+LL | fn f() -> impl Foo {
+   |           ^^^^^^^^ the trait `Foo` is not implemented for `i32`
+...
+LL |     1i32
+   |     ---- return type was inferred to be `i32` here
+   |
+help: this trait has no implementations, consider adding one
+  --> $DIR/eagerly-emit.rs:3:1
+   |
+LL | trait Foo {}
+   | ^^^^^^^^^
+
+error: expected fulfillment errors
```

Comment:
It's very strange that compiletest doesn't make you `//~ expected fulfillment errors`, though I consider that bug to be orthogonal to this PR.

Reply:
It's not a bug, `//~` only checks diagnostics that have a (non-dummy) span.
You should be able to do `// error-pattern: expected fulfillment errors` (while keeping the other `//~`s).